### ISSUE TITLE
Question about Youtube audio only

### ISSUE THREAD

**Issue by Newcool1230** [cyfung1031/userscript-supports#50]
Hey there, 

I was wondering if you could possible walk through how you are achieving audio only after the recent youtube update, where they are merging both audio and video into one `https://*.googlevideo.com/videoplayback/*` request.

Previously audio only was achieved by blocking `mime=video` requests and only allowing `mime=audio` requests. After the update this doesn't seem to work as they are merged into one request.

Looking at your code it seems like you are updating some flags, but I can't seem to get a grasp on it. Would love to learn more about this.

**Comment by cyfung1031** [comment]
I don't know how the actual YouTube engine is working with the network requests.

There is a property `deviceIsAudioOnly` existing in YouTube coding. My script is to force it to `true` only, plus some fixes for the live streaming.

**Comment by cyfung1031** [comment]
No... This is not that simple
You need to study the JavaScript code of the YouTube. 

Anyway you can just use my script for this feature

**Comment by cyfung1031** [comment]
`deviceIsAudioOnly`

https://github.com/cyfung1031/userscript-supports/blob/f52e0a6040efc4ba80c749b543f058dc5464a7d6/484611-youtube-audio-only.user.js#L2134-L2150


I could not go through all the coding I wrote to tell you how to implement this.
You can use my userscript directly or deeply analyse YouTube's JavaScript by yourself.
